src/goto-programs/read_goto_binary.cpp

```diff
@@ -14,18 +14,207 @@ Module: Read Goto Programs
 #include <fstream>
 #include <unordered_set>
 
+#include <util/base_type.h>
+#include <util/config.h>
+#include <util/exception_utils.h>
 #include <util/message.h>
-#include <util/unicode.h>
-#include <util/tempfile.h>
 #include <util/rename_symbol.h>
-#include <util/config.h>
+#include <util/replace_symbol.h>
+#include <util/symbol.h>
+#include <util/tempfile.h>
+#include <util/unicode.h>
+
+#include <linking/linking_class.h>
 
 #include "goto_model.h"
-#include "link_goto_model.h"
 #include "read_bin_goto_object.h"
 #include "elf_reader.h"
 #include "osx_fat_reader.h"
 
+static void rename_symbols_in_function(
+  goto_functionst::goto_functiont &function,
+  irep_idt &new_function_name,
+  const rename_symbolt &rename_symbol)
+{
+  for(auto &identifier : function.parameter_identifiers)
+  {
+    auto entry = rename_symbol.expr_map.find(identifier);
+    if(entry != rename_symbol.expr_map.end())
+      identifier = entry->second;
+  }
+
+  goto_programt &program = function.body;
+
+  Forall_goto_program_instructions(iit, program)
```

Comment:
At the risk of submitting a piss-take comment -- do you really want to add a macro for-all given the other PRs of yours I have open?

Reply:
I could fix this here, but those lines are just an artefact of moving the code from `link_goto_model.cpp` to `read_goto_binary.cpp`, and I'd hence prefer to keep such changes in a separate commit.

Reply:
Fair.